File: aiguigu/src/main/scala/chapter01/Student.scala
package chapter01

import java.sql.Struct

class Student(name: String, var age: Int) {
  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
//    println(this.name + " " + this.age + " " + Student.school)  // 与上面的一样
  }
}

// 引入伴生对象
object Student{
  val school: String = "aiguigu"

  def main(args: Array[String]): Unit = {
    val alice = new Student("alice", 20)
    val bob = new Student("bob", 23)

    alice.printInfo()
    bob.printInfo()
  }
}

File: aiguigu/src/main/java/Student.java
public class Student {
    private String name;
    private Integer age;
    private static String school = "atguigu";  // static 静态，所有学生school都是一样的

    public Student(String name, Integer age){
        this.name = name;
        this.age = age;
    }

    public void printInfo(){
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }

    public static void main(String[] args) {
        Student alice = new Student("alice", 20);
        Student bob = new Student("bob", 23);
        alice.printInfo();
        bob.printInfo();
    }

}
